fe/fe-core/src/main/java/com/starrocks/load/InsertOverwriteJobRunner.java

```diff
@@ -0,0 +1,308 @@
+                job.setNewPartitionNames(info.getNewPartitionsName());
+                job.setJobState(InsertOverwriteJobState.OVERWRITE_RUNNING);
+                break;
+            case OVERWRITE_FAILED:
+                job.setJobState(InsertOverwriteJobState.OVERWRITE_FAILED);
+                LOG.info("replay insert overwrite job:{} to FAILED", job.getJobId());
+                gc();
+                break;
+            case OVERWRITE_SUCCESS:
+                job.setJobState(InsertOverwriteJobState.OVERWRITE_SUCCESS);
+                doCommit();
+                LOG.info("replay insert overwrite job:{} to SUCCESS", job.getJobId());
+                break;
+            default:
+                LOG.warn("invalid to state:{} for insert overwrite job:{}", info.getToState(), job.getJobId());
+                break;
+        }
+    }
+
+    private void transferTo(InsertOverwriteJobState state) throws Exception {
+        if (state == InsertOverwriteJobState.OVERWRITE_SUCCESS) {
+            Preconditions.checkState(job.getJobState() == InsertOverwriteJobState.OVERWRITE_RUNNING);
+        }
+        InsertOverwriteStateChangeInfo info =
+                new InsertOverwriteStateChangeInfo(job.getJobId(), job.getJobState(), state,
+                        job.getSourcePartitionNames(), job.getNewPartitionNames());
+        GlobalStateMgr.getCurrentState().getEditLog().logInsertOverwriteStateChange(info);
+        job.setJobState(state);
+        handle();
+    }
+
+    private void prepare() throws Exception {
+        Preconditions.checkState(job.getJobState() == InsertOverwriteJobState.OVERWRITE_PENDING);
+        List<Partition> sourcePartitions = job.getOriginalTargetPartitionIds().stream()
+                .map(id -> targetTable.getPartition(id)).collect(Collectors.toList());
+        List<String> sourcePartitionNames = sourcePartitions.stream().map(p -> p.getName()).collect(Collectors.toList());
+        job.setSourcePartitionNames(sourcePartitionNames);
+        job.setNewPartitionNames(sourcePartitionNames.stream().map(name -> name + postfix).collect(Collectors.toList()));
+        transferTo(InsertOverwriteJobState.OVERWRITE_RUNNING);
+    }
+
+    private void executeInsert() throws Exception {
+        long insertStartTimestamp = System.currentTimeMillis();
+        // should replan here because prepareInsert has changed the targetPartitionNames of insertStmt
+        ExecPlan newPlan = new StatementPlanner().plan(insertStmt, context);
+        stmtExecutor.handleDMLStmt(newPlan, insertStmt);
+        insertElapse = System.currentTimeMillis() - insertStartTimestamp;
+        if (context.getState().getStateType() == QueryState.MysqlStateType.ERR) {
+            LOG.warn("insert overwrite failed. error message:{}", context.getState().getErrorMessage());
+            throw new RuntimeException("execute insert failed");
+        }
+    }
+
+    private void createTempPartitions() {
```

Comment:
Can you make it in a PartitionUtils class, then others can create tempartition easily. like
`PartitionUtils.createTempPartition`

Reply:
done